src/main/java/Sprites.java

```diff
@@ -141,4 +142,36 @@ public int getScriptCountForSprite(String spriteName) {
         }
         return lengths;
     }
+    
+    /**
+     * Public accessor method for getting the number of variables 
+     * associated with a particular sprite.
+     * @param sprite the name of the sprite whose info you desire
+     * @return the number of variables associated with a sprite 
+     * @throws IOException if the sprite sought does not exist 
+     */ 
+    public int getSpriteVariableCount(String sprite) throws IOException {
+        if (spriteMap.containsKey(sprite)) {
+            countSpriteVariables(spriteMap.get(sprite));
+            return variables;
+        } else {
+            throw new IOException("You should not be searching for sprites"
+                + " that don't exist.");
+        }
+    }
+    /**
+     * Private method that attempts to find the number variables 
+     * associated with a sprite and store it as a field variable 
+     * associated with the sprite.
+     * @param sprite the json representation of the sprite object
+     *      from an sb2 object
+     */
+    private void countSpriteVariables(JSONObject sprite) {
```

Comment:
Same thing here.  I'd change this method to `public int countSpriteVariables(String spriteName)`, get rid of `public int getSpriteVariableCount(String spriteName)` and make `int variables` local to `countSpriteVariables`.  It would make things much simpler. 

That said, the code you've written accomplishes what we need.  These are just refactoring suggestions.